src/utils/file_upload.rs

```diff
@@ -420,13 +420,21 @@ fn build_debug_id(files: &SourceFiles) -> DebugId {
         if let Some(debug_id) = source_file
             .headers
             .iter()
-            .filter_map(|(k, v)| (k == "debug_id").then_some(v))
-            .next()
+            .find_map(|(k, v)| (k == "debug_id").then_some(v))
         {
             hash.update(debug_id.as_bytes());
         } else {
             hash.update(&[0u8; 16]);
         }
+
+        if let Some(sourcemap_ref) = source_file
+            .headers
```

Comment:
instead of picking individual pieces from the manifest, why not just hash all of `SourceFiles`?

You are also collecting / sorting `SourceFiles` twice with different keys here and down below in `build_artifact_bundle`, where you are actually calling out to `build_debug_id`.

I wonder if we can build this determinism guarantee into `SourceFiles` directly, so that it is guaranteed to be sorted in a deterministic way, and that it hashes in a deterministic way as well.